Tasks/NuGetCommand/nugetpublisher.ts

```diff
@@ -328,5 +328,10 @@ function shouldUseVstsNuGetPush(isInternalFeed: boolean, conflictsAllowed: boole
         return false;
     }
 
+    if (!(tl.osType() === 'Windows_NT' || !nugetExePath.trim().toLowerCase().endsWith(".exe"))) {
```

Comment:
Why do we need to check if the file ends with.exe? We can't assume that the nuget binary will always end with .exe, specifally in Unix agents. Could it be possible that a user specify a nugetExePath which doesn't end with .exe?